### ISSUE TITLE
Empty header cell in datagrid needs a title

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3464]
## Describe the bug

When using single selection, detail pane, or expandable rows, there is an empty cell in the header that has no title text. It is unclear to a screenreader what these cells are for, especially when navigating to the cells further down in the column.

## How to reproduce

- Go to the datagrid expandable rows example
- Observe that the first column header is empty.
- Using a screen reader see it simply states you are on a cell.

## Expected behavior

The cells should have meaningful text, even if it is hidden off page.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

We already have a class to hide content off screen, we can add content into these header cells and hide it with that.
